.github/workflows/run-tests.yml

```diff
@@ -145,21 +164,47 @@ jobs:
         run: |
           docker compose -f ${{ env.DOCKER_COMPOSE_FILE }} down -v
 
+
+  set-up-functional-test:
+    name: Set up functional test
+    runs-on: ubuntu-20.04
+    outputs:
+      shard_indices: ${{ steps.generate_output.outputs.shard_indices }}
+      total_shards: ${{ steps.generate_output.outputs.shards }}
+      single_test_args: ${{ steps.generate_output.outputs.single_test_args }}
+      functional_test_timeout: ${{ steps.generate_output.outputs.timeout }}
+    steps:
+      - id: generate_output
+        run: |
+          shards=3
+          timeout=30
+          if [[ "${{ inputs.run_single_functional_test }}" == "true" ]]; then
+            shards=1
+            timeout=${{ inputs.timeout_minutes }}
+            single_test_args="-test.run ${{ inputs.test_name }} -count ${{ inputs.n_runs }}"
+          fi
+          {
+            echo "shard_indices=[$(seq -s, 0 $((shards-1)))]"
+            echo "shards=$shards"
+            echo "timeout=$timeout"
+            echo "single_test_args=$single_test_args" 
+          } >> "$GITHUB_OUTPUT"
+
   functional-test:
     name: Functional test
-    needs: [misc-checks, cache-docker-images]
+    needs: [cache-docker-images, set-up-functional-test]
```

Comment:
Removed `misc-checks`?

Reply:
In the case where you're trying to just run a single test, it seems unnecessary to me to run misc-checks. It would be annoying to make this `needs` list conditional based on the "mode", so I would be ok with putting it back in, but also I'm curious why this dependency was there in the first place?

Reply:
after running this a bunch, I think I see why the dependency is good. If misc-checks fails, we don't want to kick off 12 whole jobs that are also going to fail. I'll put it back